compiler/il/OMRILOps.hpp

```diff
@@ -1137,6 +1137,20 @@ class ILOpCode
       return TR::BadILOp;
       }
 
+   static TR::ILOpCodes ternaryOpCode(TR::DataType type)
```

Comment:
Since there's an obvious set of missing data types here (float/double) I think I would prefer a more tolerant API that could be used by `Select`  (i.e. it doesn't `TR_ASSERT`)

Reply:
Hm, there is `fternary` and `dternary`...are these not implemented on all platforms?

There is also `vternary` for vector types....

Reply:
I've left out `fternary`, `dternary` and `vternary` out because they are not implemented on all platforms. `fternary` and `dternary` are not implemented on x86, and `vternary` is only implemented on Z. I will update `ternaryOpCode` to return those opcodes anyway and not assert.

Reply:
I'm ok with this choice as an initial design point, but please open an issue for us to work out the (IMO) better design point "do the best you can on each platform" rather than "lowest common denominator".